src/components/BodyWeight/widgets/FilterButtons.tsx

```diff
@@ -0,0 +1,67 @@
+// components/BodyWeight/widgets/FilterButtons.tsx
+
+import { Button, ButtonGroup } from "@mui/material";
+import { useTheme } from '@mui/material/styles';
+
+export type FilterType = 'lastYear' | 'lastHalfYear' | 'lastMonth' | 'lastWeek' | '';
+
+export interface FilterButtonsProps {
+    currentFilter: FilterType;
+    onFilterChange: (newFilter: FilterType) => void;
+}
+
+export const FilterButtons = ({ currentFilter, onFilterChange }: FilterButtonsProps) => {
+    const theme = useTheme();
+
+    // Won't call onFilterChange if the filter stays the same 
+    const handleFilterChange = (newFilter: FilterType) => {
+        if (currentFilter !== newFilter) {
+            onFilterChange(newFilter);
+        }
+    };
+
+    return (
+        <ButtonGroup variant="outlined" sx={{ mb: 2 }}>
+            <Button
+                onClick={() => handleFilterChange('')}
+                color={currentFilter === '' ? 'primary' : 'inherit'}
+                variant={currentFilter === '' ? 'contained' : 'outlined'}
+                sx={{ fontFamily: theme.typography.fontFamily }} 
+            >
+                All
```

Comment:
can you make the labels translatable? Just add them to `public/locales/en/translation.json` and then you can just use them with `{t('key')}`. Just add them to the root, these could potentially be also used soemwhere else in the application

Reply:
Of course! I've updated the code based on your feedback.